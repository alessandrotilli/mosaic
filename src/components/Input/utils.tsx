import React from "react";
import { InputAdornment as MUIInputAdornment } from "@material-ui/core";

import { IInputAdornment } from "../../types/Input";
import IconButton from "../IconButton";

export const getAdornment = (adornment?: IInputAdornment) => {
  if (!adornment) {
    return;
  }

  const { icon, onClick } = adornment;
  return (
    <MUIInputAdornment position="end">
      <IconButton disabled={!onClick} icon={icon} onClick={onClick!} size="small" />
    </MUIInputAdornment>
  );
};
