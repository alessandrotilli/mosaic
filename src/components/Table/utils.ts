export const COLUMN_CHECKBOX_PATH = "checkbox-selection";
export const COLUMN_ROW_ACTIONS_PATH = "row-actions";

export const TOOLBAR_HEIGHT = 64;
export const TOOLBAR_HEIGHT_MOBILE = 56;

export const PAGINATION_TOOLBAR_HEIGHT = 64;
