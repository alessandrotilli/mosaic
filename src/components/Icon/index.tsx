import React, { cloneElement, FC, PropsWithChildren, ReactElement, useCallback, useMemo } from "react";
import { ClassNames } from "@emotion/react";
import { keyframes, Skeleton as MUISkeleton } from "@mui/material";

import { IIcon, IIconDimensions, IRenderedIcon } from "../../types/Icon";
import { logWarn } from "../../utils/logger";
import Adornment from "../Adornment";

import { iconsCatalog } from "./utils";

export const DATA_CY_DEFAULT = "icon";

const ICON_DIMENSIONS: IIconDimensions = {
  small: 20,
  medium: 24,
  large: 35,
};

const Icon: FC<PropsWithChildren<IIcon>> = ({
  badge,
  children,
  dataCy = DATA_CY_DEFAULT,
  forwarded = {},
  loading = false,
  name,
  rotate = false,
  size = "medium",
  style: externalStyle,
  tooltip,
}) => {
  const dimensions = useMemo(() => {
    const dimension = ICON_DIMENSIONS[size];
    return { height: dimension, width: dimension };
  }, [size]);

  const props = useMemo(() => {
    let props: IRenderedIcon = {
      ...forwarded,
      "data-cy": dataCy,
      fontSize: size,
    };

    if (externalStyle) {
      props = { ...props, style: { ...externalStyle } };
    }

    if (children) {
      props = { ...props, style: { ...props.style, ...dimensions } };
    }

    return props;
  }, [children, dataCy, dimensions, externalStyle, forwarded, size]);

  const rotateKeyframes = useMemo(
    () => keyframes`
      from: {
        transform: "rotate(0deg)";
      }
      to: {
        transform: "rotate(360deg)";
      }
  `,
    []
  );

  const renderWithAnimation = useCallback(
    (element: any) => {
      return (
        <ClassNames>
          {({ css }) => {
            let animation = "none";
            if (rotate) {
              animation = css`
                ${rotateKeyframes} 2s linear infinite
              `;
            }

            let className = css`
              animation: ${animation};
            `;

            if (forwarded.className) {
              className = forwarded.className;
            }

            return (
              <Adornment badge={badge} tooltip={tooltip}>
                {cloneElement(element as ReactElement<any>, { ...props, className })}
              </Adornment>
            );
          }}
        </ClassNames>
      );
    },
    [badge, forwarded, props, rotate, rotateKeyframes, tooltip]
  );

  if (loading) {
    return <MUISkeleton variant="rectangular" style={{ ...dimensions }} />;
  }

  if (children) {
    return renderWithAnimation(children);
  }

  if (!name) {
    logWarn("Icon", "Skip rendering, both children and name are not set");
    return null;
  }

  const icon = iconsCatalog[name];
  return renderWithAnimation(icon);
};

export default Icon;
