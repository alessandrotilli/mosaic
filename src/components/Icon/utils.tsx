import React from "react";
import MUIAccountCircleIcon from "@material-ui/icons/AccountCircle";
import MUIAddIcon from "@material-ui/icons/Add";
import MUIAppsIcon from "@material-ui/icons/Apps";
import MUIArrowUpwardIcon from "@material-ui/icons/ArrowUpward";
import MUIBlockIcon from "@material-ui/icons/Block";
import MUIBrightness4Icon from "@material-ui/icons/Brightness4";
import MUIBusinessIcon from "@material-ui/icons/Business";
import MUICheckIcon from "@material-ui/icons/Check";
import MUICheckBoxIcon from "@material-ui/icons/CheckBox";
import MUICheckBoxOutlineBlankIcon from "@material-ui/icons/CheckBoxOutlineBlank";
import MUIChevronLeftIcon from "@material-ui/icons/ChevronLeft";
import MUIChevronRightIcon from "@material-ui/icons/ChevronRight";
import MUICloseIcon from "@material-ui/icons/Close";
import MUICropIcon from "@material-ui/icons/Crop";
import MUIDeleteIcon from "@material-ui/icons/Delete";
import MUIDetailsIcon from "@material-ui/icons/Details";
import MUIDraftsIcon from "@material-ui/icons/Drafts";
import MUIEditIcon from "@material-ui/icons/Edit";
import MUIErrorIcon from "@material-ui/icons/Error";
import MUIFilterListIcon from "@material-ui/icons/FilterList";
import MUIFirstPageIcon from "@material-ui/icons/FirstPage";
import MUIGroupWorkIcon from "@material-ui/icons/GroupWork";
import MUIHomeIcon from "@material-ui/icons/Home";
import MUIKeyboardArrowDownIcon from "@material-ui/icons/KeyboardArrowDown";
import MUIKeyboardArrowUpIcon from "@material-ui/icons/KeyboardArrowUp";
import MUILanguageIcon from "@material-ui/icons/Language";
import MUILastPageIcon from "@material-ui/icons/LastPage";
import MUIListIcon from "@material-ui/icons/List";
import MUIMailIcon from "@material-ui/icons/Mail";
import MUIMenuIcon from "@material-ui/icons/Menu";
import MUINotificationsIcon from "@material-ui/icons/Notifications";
import MUIOpenInNew from "@material-ui/icons/OpenInNew";
import MUIPauseIcon from "@material-ui/icons/Pause";
import MUIPlayArrowIcon from "@material-ui/icons/PlayArrow";
import MUIPlaylistAddCheckIcon from "@material-ui/icons/PlaylistAddCheck";
import MUIPowerSettingsNewIcon from "@material-ui/icons/PowerSettingsNew";
import MUIPrintIcon from "@material-ui/icons/Print";
import MUIRefreshIcon from "@material-ui/icons/Refresh";
import MUISaveIcon from "@material-ui/icons/Save";
import MUIScheduleIcon from "@material-ui/icons/Schedule";
import MUISearchIcon from "@material-ui/icons/Search";
import MUISendIcon from "@material-ui/icons/Send";
import MUISettingsIcon from "@material-ui/icons/Settings";
import MUISignalWifi1BarOutlinedIcon from "@material-ui/icons/SignalWifi1BarOutlined";
import MUITimelineIcon from "@material-ui/icons/Timeline";
import MUIVisibilityIcon from "@material-ui/icons/Visibility";
import MUIVisibilityOffIcon from "@material-ui/icons/VisibilityOff";
import MUIWarningIcon from "@material-ui/icons/Warning";
import MUIWifiOffOutlinedIcon from "@material-ui/icons/WifiOffOutlined";
import MUIWifiOutlinedIcon from "@material-ui/icons/WifiOutlined";

import { Icons, IIconsCatalog } from "../../types/Icon";

export const iconsCatalog: IIconsCatalog = {
  [Icons.account]: <MUIAccountCircleIcon />,
  [Icons.add]: <MUIAddIcon />,
  [Icons.apps]: <MUIAppsIcon />,
  [Icons.block]: <MUIBlockIcon />,
  [Icons.brightness]: <MUIBrightness4Icon />,
  [Icons.business]: <MUIBusinessIcon />,
  [Icons.chart_timeline]: <MUITimelineIcon />,
  [Icons.check]: <MUICheckIcon />,
  [Icons.checkbox]: <MUICheckBoxIcon />,
  [Icons.checkbox_empty]: <MUICheckBoxOutlineBlankIcon />,
  [Icons.clock]: <MUIScheduleIcon />,
  [Icons.close]: <MUICloseIcon />,
  [Icons.connection_slow]: <MUISignalWifi1BarOutlinedIcon />,
  [Icons.crop]: <MUICropIcon />,
  [Icons.delete]: <MUIDeleteIcon />,
  [Icons.details]: <MUIDetailsIcon />,
  [Icons.down]: <MUIKeyboardArrowDownIcon />,
  [Icons.edit]: <MUIEditIcon />,
  [Icons.error]: <MUIErrorIcon />,
  [Icons.filter]: <MUIFilterListIcon />,
  [Icons.first]: <MUIFirstPageIcon />,
  [Icons.group]: <MUIGroupWorkIcon />,
  [Icons.home]: <MUIHomeIcon />,
  [Icons.language]: <MUILanguageIcon />,
  [Icons.last]: <MUILastPageIcon />,
  [Icons.left]: <MUIChevronLeftIcon />,
  [Icons.list_check]: <MUIPlaylistAddCheckIcon />,
  [Icons.list]: <MUIListIcon />,
  [Icons.mail_open]: <MUIDraftsIcon />,
  [Icons.mail]: <MUIMailIcon />,
  [Icons.menu]: <MUIMenuIcon />,
  [Icons.notifications]: <MUINotificationsIcon />,
  [Icons.open_new]: <MUIOpenInNew />,
  [Icons.pause]: <MUIPauseIcon />,
  [Icons.play]: <MUIPlayArrowIcon />,
  [Icons.power]: <MUIPowerSettingsNewIcon />,
  [Icons.print]: <MUIPrintIcon />,
  [Icons.refresh]: <MUIRefreshIcon />,
  [Icons.right]: <MUIChevronRightIcon />,
  [Icons.save]: <MUISaveIcon />,
  [Icons.search]: <MUISearchIcon />,
  [Icons.send]: <MUISendIcon />,
  [Icons.settings]: <MUISettingsIcon />,
  [Icons.up_arrow]: <MUIArrowUpwardIcon />,
  [Icons.up]: <MUIKeyboardArrowUpIcon />,
  [Icons.visibility]: <MUIVisibilityIcon />,
  [Icons.visibility_off]: <MUIVisibilityOffIcon />,
  [Icons.warn]: <MUIWarningIcon />,
  [Icons.wifi_off]: <MUIWifiOffOutlinedIcon />,
  [Icons.wifi_on]: <MUIWifiOutlinedIcon />,
};
