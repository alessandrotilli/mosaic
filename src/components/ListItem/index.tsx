import React, { CSSProperties, FC, useCallback, useMemo } from "react";
import {
  ListItem as MUIListItem,
  ListItemIcon as MUIListItemIcon,
  ListItemText as MUIListItemText,
} from "@material-ui/core";
import { Skeleton as MUISkeleton } from "@material-ui/lab";

import { IListItem } from "../../types/ListItem";
import { getComposedDataCy, suppressEvent } from "../../utils";
import IconWrapper from "../IconWrapper";

export const DATA_CY_DEFAULT = "list-item";

export const SUBPARTS_MAP = {
  icon: {
    label: "Icon",
  },
  content: {
    label: "Content (with loading = true, adds '-loading')",
    value: (loading = false) => `content${loading ? "-loading" : ""}`,
  },
};

const ListItem: FC<IListItem> = ({
  children,
  content,
  dataCy = DATA_CY_DEFAULT,
  dense = false,
  icon,
  loading = false,
  onClick: externalOnClick,
  selected = false,
  style,
}) => {
  const baseStyle: CSSProperties = useMemo(
    () => ({ cursor: !loading && externalOnClick ? "pointer" : "default" }),
    [externalOnClick, loading]
  );

  const onClick = useCallback(
    (event) => {
      suppressEvent(event);
      if (loading) {
        return;
      }

      externalOnClick && externalOnClick();
    },
    [externalOnClick, loading]
  );

  return (
    <MUIListItem
      data-cy={dataCy}
      dense={dense}
      onClick={onClick}
      selected={!loading && selected}
      style={{ ...baseStyle, ...style }}
    >
      {icon && (
        <MUIListItemIcon>
          <IconWrapper dataCy={getComposedDataCy(dataCy, SUBPARTS_MAP.icon)} icon={icon} loading={loading} />
        </MUIListItemIcon>
      )}
      <MUIListItemText data-cy={getComposedDataCy(dataCy, SUBPARTS_MAP.content, loading)} disableTypography>
        {loading ? <MUISkeleton /> : content || children}
      </MUIListItemText>
    </MUIListItem>
  );
};

export default ListItem;
