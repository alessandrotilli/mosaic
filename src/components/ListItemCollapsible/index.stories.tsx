import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";

import { getAllComposedDataCy } from "../../utils";
import getDocsPage from "../../utils/stories";

import ListItemCollapsible, { DATA_CY_DEFAULT, SUBPARTS_MAP } from ".";

export default {
  title: "Display/ListItemCollapsible",
  component: ListItemCollapsible,
  parameters: {
    docs: {
      ...getDocsPage({
        basedOn: {
          label: "MUI Collapse Component",
          url: "https://v4.mui.com/components/transitions/#collapse",
        },
        component: "ListItemCollapsible",
        e2eTestInfo: {
          dataCyDefault: DATA_CY_DEFAULT,
          subpartsSuffixes: getAllComposedDataCy(SUBPARTS_MAP),
        },
      }),
    },
  },
} as ComponentMeta<typeof ListItemCollapsible>;

const Template: ComponentStory<typeof ListItemCollapsible> = (args) => (
  <ListItemCollapsible {...args} dataCy={DATA_CY_DEFAULT} />
);

export const Primary = Template.bind({});
Primary.args = {
  content: "Collapsible Content",
  header: "Header",
};

export const Dense = Template.bind({});
Dense.args = {
  ...Primary.args,
  dense: true,
};

export const Loading = Template.bind({});
Loading.args = {
  ...Primary.args,
  loading: true,
};

export const Open = Template.bind({});
Open.args = {
  ...Primary.args,
  open: true,
};

export const Selected = Template.bind({});
Selected.args = {
  ...Primary.args,
  selected: true,
};

// export const Basic = () => <ListItemCollapsible header={<Typography>Collapsible List Item</Typography>} />;

// export const CustomStyle = () => (
//   <ListItemCollapsible
//     header={<Typography>Custom Style</Typography>}
//     style={{ backgroundColor: "red", color: "white" }}
//   />
// );

// export const Dense = () => <ListItemCollapsible dense header={<Typography>Collapsible List Item</Typography>} />;

// export const Loading = () => <ListItemCollapsible loading header={<Typography>Collapsible List Item</Typography>} />;

// export const Selected = () => <ListItemCollapsible selected header={<Typography>Collapsible List Item</Typography>} />;

// export const Open = () => (
//   <ListItemCollapsible open selected header={<Typography>Collapsible List Item</Typography>}>
//     <div style={{ border: "1px solid #ccc", padding: "16px" }}>
//       <Typography children="Collapsible content is now expanded" />
//     </div>
//   </ListItemCollapsible>
// );
