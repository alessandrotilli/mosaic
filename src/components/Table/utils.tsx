import React, { forwardRef, Fragment } from "react";
import { Action as MTActionType, Column as MTColumnType, Options as MTOptionsType } from "material-table";
import { Icons, IconSize } from "../../types/Icon";
import { TableActionPosition, ITableAction, ITableColumn } from "../../types/Table";
import Button from "../Button";
import Icon from "../Icon";
import IconButton from "../IconButton";
import { getDataCyForSubComponent, suppressEvent } from "../../utils";

// For default props refer to Options section here https://material-table.com/#/docs/all-props
export const DEFAULT_TABLE_OPTIONS: MTOptionsType = {
  debounceInterval: 500,
  draggable: false,
  emptyRowsWhenPaging: false,
};

export const actionAdapter = (action: ITableAction): MTActionType<object> => {
  const {
    callback,
    label,
    disabled = false,
    hidden = false,
    icon = undefined,
    position = TableActionPosition.default,
  } = action;
  return {
    disabled,
    hidden,
    icon: `${label}$${icon}`,
    onClick: (event, data) => {
      if (event) {
        suppressEvent(event);
      }
      callback(data);
    },
    position,
  };
};

export const actionComponentAdapter = (props: any, dataCy: string) => {
  const { data, action } = props;
  const { disabled, hidden, icon, position, onClick }: MTActionType<object> = action;
  if (hidden) {
    return null;
  }

  const displaysButton = position === TableActionPosition.default;
  const labelAndIconName = icon as string;
  const label = labelAndIconName.split("$")[0];
  const iconName = labelAndIconName.split("$")[1] as Icons;
  return (
    <Fragment>
      {displaysButton && (
        <Fragment>
          <span>&nbsp;&nbsp;&nbsp;&nbsp;</span>
          <Button
            dataCy={getDataCyForSubComponent(dataCy, `action-${label}`)}
            disabled={disabled}
            icon={!iconName ? undefined : { name: iconName }}
            label={label}
            onClick={() => onClick(undefined, data)}
          />
        </Fragment>
      )}
      {!displaysButton && (
        <Fragment>
          <IconButton
            dataCy={getDataCyForSubComponent(dataCy, `action-${label}`)}
            disabled={disabled}
            icon={iconName}
            onClick={() => onClick(undefined, data)}
            size={IconSize.small}
          />
          <span>&nbsp;&nbsp;</span>
        </Fragment>
      )}
    </Fragment>
  );
};

export const columnAdapter = (column: ITableColumn): MTColumnType<object> => {
  const { label, path, render, width } = column;
  return {
    ...(!!width && ({ width } as object)),
    field: path,
    render,
    title: label,
  };
};

export const iconAdapter = (name: Icons, size = IconSize.default) =>
  forwardRef((props: any, ref: any) => {
    return <Icon forwarded={{ ...props, ref }} name={name} size={size} />;
  });
