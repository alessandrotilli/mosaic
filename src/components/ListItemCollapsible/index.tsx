import React, { FC, memo, PropsWithChildren, useMemo } from "react";
import { Collapse as MUICollapse } from "@mui/material";

import { Icons } from "../../types/Icon";
import { IListItemCollapsible } from "../../types/ListItem";
import { getComposedDataCy } from "../../utils";
import ListItem from "../ListItem";

export const DATA_CY_DEFAULT = "list-item-collapsible";

export const SUBPARTS_MAP = {
  header: {
    label: "Header",
  },
  collapse: {
    label: "Collapse",
  },
};

const ListItemCollapsible: FC<PropsWithChildren<IListItemCollapsible>> = ({
  children,
  content,
  dataCy = DATA_CY_DEFAULT,
  dense = false,
  header,
  loading = false,
  onClick,
  open = false,
  openTimeout = "auto",
  selected = false,
  style,
  unmountContent = false,
}) => {
  const icon = useMemo(() => (open ? Icons.up : Icons.down), [open]);

  const collapsibleContent = useMemo(() => content || children, [content, children]);

  return (
    <>
      <ListItem
        dataCy={getComposedDataCy(dataCy, SUBPARTS_MAP.header)}
        dense={dense}
        icon={icon}
        loading={loading}
        onClick={onClick}
        selected={selected}
        style={style}
      >
        {header}
      </ListItem>
      <MUICollapse
        data-cy={getComposedDataCy(dataCy, SUBPARTS_MAP.collapse)}
        in={open}
        timeout={openTimeout}
        unmountOnExit={unmountContent}
      >
        {collapsibleContent}
      </MUICollapse>
    </>
  );
};

export default memo(ListItemCollapsible);
