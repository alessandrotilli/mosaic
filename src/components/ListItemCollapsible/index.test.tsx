import React from "react";
import renderer from "react-test-renderer";
import { IListItemCollapsible } from "../../types/ListItem";
import { getTestable } from "../../utils/tests";
import ListItemCollapsible, { DATA_CY_DEFAULT } from ".";
import { Icons } from "../../types/Icon";

const defaultProps: IListItemCollapsible = {
  header: <span>Collapsible Header</span>,
};

// TODO: unify dataCy generation
const getListItemCollapsibleTestable = (props?: IListItemCollapsible, dataCy = DATA_CY_DEFAULT, domNode = "li") =>
  getTestable(ListItemCollapsible, { dataCy, domNode, props: { ...defaultProps, ...props } });

describe("ListItemCollapsible test suite:", () => {
  it("default", () => {
    const { element, wrapper } = getListItemCollapsibleTestable({ ...defaultProps }, `${DATA_CY_DEFAULT}-header`);
    expect(wrapper).toHaveLength(1);
    const icon = wrapper.find(`Icon[dataCy='${DATA_CY_DEFAULT}-header-icon']`);
    expect(icon.prop("name")).toEqual(Icons.down);

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  it("dataCy", () => {
    const { element, wrapper } = getListItemCollapsibleTestable({ ...defaultProps, dataCy: "custom" }, "custom-header");
    expect(wrapper).toHaveLength(1);

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  it("listItem variants", () => {
    const { element, wrapper } = getListItemCollapsibleTestable(
      { ...defaultProps, dense: true, loading: true, selected: true },
      `${DATA_CY_DEFAULT}-header`
    );
    expect(wrapper.hasClass("MuiListItem-dense"));
    expect(wrapper.hasClass("MuiListItem-selected"));
    const contentWrapper = wrapper.find(`div[data-cy='${DATA_CY_DEFAULT}-header-content']`);
    expect(contentWrapper).toHaveLength(0);
    const loadingContentWrapper = wrapper.find(`div[data-cy='${DATA_CY_DEFAULT}-header-content-loading']`);
    expect(loadingContentWrapper).toHaveLength(1);

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  it("open", () => {
    const { element: header, wrapper: headerWrapper } = getListItemCollapsibleTestable(
      { ...defaultProps, open: true },
      `${DATA_CY_DEFAULT}-header`
    );
    const icon = headerWrapper.find(`Icon[dataCy='${DATA_CY_DEFAULT}-header-icon']`);
    expect(icon.prop("name")).toEqual(Icons.up);

    const { element: collapsible, wrapper: collapsibleWrapper } = getListItemCollapsibleTestable(
      { ...defaultProps, open: true },
      `${DATA_CY_DEFAULT}-collapse`,
      "div"
    );
    expect(collapsibleWrapper).toHaveLength(1);
    expect(collapsibleWrapper.hasClass("MuiCollapse-entered"));

    const snapshotHeaderWrapper = renderer.create(header).toJSON();
    expect(snapshotHeaderWrapper).toMatchSnapshot();
    const snapshotCollapsibleWrapper = renderer.create(collapsible).toJSON();
    expect(snapshotCollapsibleWrapper).toMatchSnapshot();
  });

  it("open variants", () => {
    const { element } = getListItemCollapsibleTestable(
      { ...defaultProps, open: true, openTimeout: 5, unmountContent: true },
      `${DATA_CY_DEFAULT}-collapse`,
      "div"
    );

    const snapshotHeaderWrapper = renderer.create(element).toJSON();
    expect(snapshotHeaderWrapper).toMatchSnapshot();
  });
});
