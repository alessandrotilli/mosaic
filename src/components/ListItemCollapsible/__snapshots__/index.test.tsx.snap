// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemCollapsible test suite: dataCy 1`] = `
Array [
  <li
    className="MuiListItem-root MuiListItem-gutters"
    data-cy="custom-header"
    disabled={false}
    onClick={[Function]}
    style={
      Object {
        "cursor": "default",
      }
    }
  >
    <div
      className="MuiListItemIcon-root"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        data-cy="custom-header-icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
        />
      </svg>
    </div>
    <div
      className="MuiListItemText-root"
      data-cy="custom-header-content"
    >
      <span>
        Collapsible Header
      </span>
    </div>
  </li>,
  <div
    className="MuiCollapse-container MuiCollapse-hidden"
    data-cy="custom-collapse"
    style={
      Object {
        "minHeight": "0px",
      }
    }
  >
    <div
      className="MuiCollapse-wrapper"
    >
      <div
        className="MuiCollapse-wrapperInner"
      />
    </div>
  </div>,
]
`;

exports[`ListItemCollapsible test suite: default 1`] = `
Array [
  <li
    className="MuiListItem-root MuiListItem-gutters"
    data-cy="list-item-collapsible-header"
    disabled={false}
    onClick={[Function]}
    style={
      Object {
        "cursor": "default",
      }
    }
  >
    <div
      className="MuiListItemIcon-root"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        data-cy="list-item-collapsible-header-icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
        />
      </svg>
    </div>
    <div
      className="MuiListItemText-root"
      data-cy="list-item-collapsible-header-content"
    >
      <span>
        Collapsible Header
      </span>
    </div>
  </li>,
  <div
    className="MuiCollapse-container MuiCollapse-hidden"
    data-cy="list-item-collapsible-collapse"
    style={
      Object {
        "minHeight": "0px",
      }
    }
  >
    <div
      className="MuiCollapse-wrapper"
    >
      <div
        className="MuiCollapse-wrapperInner"
      />
    </div>
  </div>,
]
`;

exports[`ListItemCollapsible test suite: listItem variants 1`] = `
Array [
  <li
    className="MuiListItem-root MuiListItem-dense MuiListItem-gutters"
    data-cy="list-item-collapsible-header"
    disabled={false}
    onClick={[Function]}
    style={
      Object {
        "cursor": "default",
      }
    }
  >
    <div
      className="MuiListItemIcon-root"
    >
      <span
        className="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
        style={
          Object {
            "height": 24,
            "width": 24,
          }
        }
      />
    </div>
    <div
      className="MuiListItemText-root MuiListItemText-dense"
      data-cy="list-item-collapsible-header-content-loading"
    >
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
    </div>
  </li>,
  <div
    className="MuiCollapse-container MuiCollapse-hidden"
    data-cy="list-item-collapsible-collapse"
    style={
      Object {
        "minHeight": "0px",
      }
    }
  >
    <div
      className="MuiCollapse-wrapper"
    >
      <div
        className="MuiCollapse-wrapperInner"
      />
    </div>
  </div>,
]
`;

exports[`ListItemCollapsible test suite: open 1`] = `
Array [
  <li
    className="MuiListItem-root MuiListItem-gutters"
    data-cy="list-item-collapsible-header"
    disabled={false}
    onClick={[Function]}
    style={
      Object {
        "cursor": "default",
      }
    }
  >
    <div
      className="MuiListItemIcon-root"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        data-cy="list-item-collapsible-header-icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z"
        />
      </svg>
    </div>
    <div
      className="MuiListItemText-root"
      data-cy="list-item-collapsible-header-content"
    >
      <span>
        Collapsible Header
      </span>
    </div>
  </li>,
  <div
    className="MuiCollapse-container MuiCollapse-entered"
    data-cy="list-item-collapsible-collapse"
    style={
      Object {
        "minHeight": "0px",
      }
    }
  >
    <div
      className="MuiCollapse-wrapper"
    >
      <div
        className="MuiCollapse-wrapperInner"
      />
    </div>
  </div>,
]
`;

exports[`ListItemCollapsible test suite: open 2`] = `
Array [
  <li
    className="MuiListItem-root MuiListItem-gutters"
    data-cy="list-item-collapsible-header"
    disabled={false}
    onClick={[Function]}
    style={
      Object {
        "cursor": "default",
      }
    }
  >
    <div
      className="MuiListItemIcon-root"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        data-cy="list-item-collapsible-header-icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z"
        />
      </svg>
    </div>
    <div
      className="MuiListItemText-root"
      data-cy="list-item-collapsible-header-content"
    >
      <span>
        Collapsible Header
      </span>
    </div>
  </li>,
  <div
    className="MuiCollapse-container MuiCollapse-entered"
    data-cy="list-item-collapsible-collapse"
    style={
      Object {
        "minHeight": "0px",
      }
    }
  >
    <div
      className="MuiCollapse-wrapper"
    >
      <div
        className="MuiCollapse-wrapperInner"
      />
    </div>
  </div>,
]
`;

exports[`ListItemCollapsible test suite: open variants 1`] = `
Array [
  <li
    className="MuiListItem-root MuiListItem-gutters"
    data-cy="list-item-collapsible-header"
    disabled={false}
    onClick={[Function]}
    style={
      Object {
        "cursor": "default",
      }
    }
  >
    <div
      className="MuiListItemIcon-root"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        data-cy="list-item-collapsible-header-icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z"
        />
      </svg>
    </div>
    <div
      className="MuiListItemText-root"
      data-cy="list-item-collapsible-header-content"
    >
      <span>
        Collapsible Header
      </span>
    </div>
  </li>,
  <div
    className="MuiCollapse-container MuiCollapse-entered"
    data-cy="list-item-collapsible-collapse"
    style={
      Object {
        "minHeight": "0px",
      }
    }
  >
    <div
      className="MuiCollapse-wrapper"
    >
      <div
        className="MuiCollapse-wrapperInner"
      />
    </div>
  </div>,
]
`;
